### PR TITLE
ADD Entry resource URL prefix parameter to RDBMS commons and some SQL connectors

### LINKED ISSUE #49: [FEATURE] Override resource URL
**What would you like to be added**:

An optional way to set the resource URL rather than using the connection string.

**Why is this needed**:

I have used the connector in two ways so far:

1. Within a k8s cluster, connecting via pgbouncer
2. From a local docker container, using port forwarding to a cloud sql proxy.

With either, the resource URL on the created table resource in Data Catalog is set to the connection string, which is either the pgbouncer service or localhost.

image

This isn't reflective of where the data is stored. It would be useful to be able to specify that the resource URL is (for example) `project-name/db-instance`.

### PR DESCRIPTION
**Pull request by mesmacosta**
**- What I did**

- Added a new parameter to RDBMS connectors, so users can specify the entry resource URL prefix.
- Added it to MySQL, PostgreSQL, Greenplum, Redshift and SQLServer connectors.
- Improved the way the linkedResource is built for Tables and Table containers.
- Improved logging for config classes.
- Added the metadata enricher to All SQL connectors.
- Updated the `__init__` files to use relative imports.

**- How I did it**
The new parameter `datacatalog-entry-resource-url-prefix` overrides the hostname as the entry resource URL prefix when informed:
```
[datacatalog_entry_resource_ur_prefix]/database/table

or when empty
[rbms_host]/database/table
```

**- How to verify it**
Run MySQL, PostgreSQL, Greenplum, Redshift and SQLServer connectors with the new parameter.

**- Description for the changelog**
ADD Entry resource URL prefix parameter to RDBMS commons and some SQL connectors, fixes #49

### CODE REVIEW

File: google-datacatalog-sqlserver-connector/src/google/datacatalog_connectors/sqlserver/datacatalog_cli.py [hunk 1]
```diff
@@ -69,6 +69,9 @@ def _parse_args(self, argv):
         parser.add_argument('--datacatalog-entry-group-id',
                             help='Entry group ID to be used for your Google '
                             'Cloud Datacatalog')
+        parser.add_argument('--datacatalog-entry-resource-url-prefix',
+                            help='Entry resource URL prefix'
```
Comment (ricardolsmendes) [inline google-datacatalog-sqlserver-connector/src/google/datacatalog_connectors/sqlserver/datacatalog_cli.py:73]: Same question about the help string...

**Review by ricardolsmendes — CHANGES_REQUESTED** [review]
Hi @mesmacosta, I've just left a few comments. Please take a look and let me know your thoughts. Thanks!

File: google-datacatalog-rdbms-connector/src/google/datacatalog_connectors/rdbms/datacatalog_cli.py [hunk 2]
```diff
@@ -103,6 +106,14 @@ def _get_entry_group_id(self, args):
     def _parse_args(self, argv):
         pass
 
+    def _get_entry_resource_url_prefix(self, args):
+        # try/except clause for backwards compatibility with
```
[the remark this answers is not in the document]
Comment (mesmacosta) [reply google-datacatalog-rdbms-connector/src/google/datacatalog_connectors/rdbms/datacatalog_cli.py:110]: Makes sense it sounds more like as a fallback treatment. I will change the comment.

File: google-datacatalog-mysql-connector/src/google/datacatalog_connectors/mysql_/datacatalog_cli.py [hunk 3]
```diff
@@ -66,6 +66,9 @@ def _parse_args(self, argv):
         parser.add_argument('--datacatalog-entry-group-id',
                             help='Entry group ID to be used for your Google '
                             'Cloud Datacatalog')
+        parser.add_argument('--datacatalog-entry-resource-url-prefix',
+                            help='Entry resource URL prefix'
```
[the remark this answers is not in the document]
Comment (mesmacosta) [reply google-datacatalog-mysql-connector/src/google/datacatalog_connectors/mysql_/datacatalog_cli.py:70]: Thanks for the catch, it was certainly missing.

**Review by ricardolsmendes — APPROVED** [review]
LGTM. Thanks for working on this, @mesmacosta!

**Pull request merged**